mne/filter.py

```diff
@@ -1830,17 +1831,24 @@ def float_array(c):
         elif not isinstance(filter_length, integer_types):
             raise ValueError('filter_length must be a str, int, or None, got '
                              '%s' % (type(filter_length),))
-    if method != 'fir':
-        filter_length = len_x
+
     if phase == 'zero' and method == 'fir':
         filter_length += (filter_length % 2 == 0)
     if filter_length <= 0:
         raise ValueError('filter_length must be positive, got %s'
                          % (filter_length,))
-    if filter_length > len_x:
-        warn('filter_length (%s) is longer than the signal (%s), '
-             'distortion is likely. Reduce filter length or filter a '
-             'longer signal.' % (filter_length, len_x))
+
+    # If we have data supplied, do a sanity check
+    if x is not None:
+        x = np.asanyarray(x)
+        len_x = x.shape[-1]
+        if method != 'fir':
+            filter_length = len_x
```

Comment:
This should be fine ... but I'm a bit unsure about line 1845 and 1846. Will the case ever happen at all? When method!='fir', then method=='iir' - and in that case `_triage_filter_params` is never called.